### PR TITLE
[core] Support for mistral embeddings

### LINKED ISSUE #5182: [Qdrant] Add support for Mistral embedder

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes: https://github.com/dust-tt/dust/issues/5182

Adds support for Mistral embeddings with `mistral-embed`

## Risk

None, not used yet.

## Deploy Plan

- deploy `core`

### CODE REVIEW

File: core/src/providers/mistral.rs [hunk 1]
```diff
@@ -979,8 +1138,7 @@ impl Provider for MistralProvider {
         utils::info(
             "To use Mistral AI's models, you must set the environment variable `MISTRAL_API_KEY`.",
         );
-        // TODO(flav): Update the link.
-        // utils::info("Your API key can be found at `https://platform.openai.com/account/api-keys`.");
+        utils::info("Your API key can be found at `https://console.mistral.ai/api-keys/`.");
```
Comment (flvndvd) [inline core/src/providers/mistral.rs:1141]: \o/

File: core/src/providers/mistral.rs [hunk 2]
```diff
@@ -959,6 +957,167 @@ impl LLM for MistralAILLM {
...
+#[async_trait]
+impl Embedder for MistralEmbedder {
+    fn id(&self) -> String {
+        self.id.clone()
+    }
+
+    async fn initialize(&mut self, credentials: Credentials) -> Result<()> {
+        if !(vec!["mistral-embed"].contains(&self.id.as_str())) {
+            return Err(anyhow!(
+                "Unexpected embedder model id (`{}`) for provider `openai`",
```
Comment (flvndvd) [inline core/src/providers/mistral.rs:1004]:
```suggestion
                "Unexpected embedder model id (`{}`) for provider `mistral`",
```

File: core/src/providers/mistral.rs [hunk 3]
```diff
@@ -959,6 +957,167 @@ impl LLM for MistralAILLM {
...
+                            factor: 1,
+                            retries: 1,
+                        }),
+                    }),
+                }
+            }
+        }?;
+
+        assert!(embeddings.data.len() > 0);
+        // println!("EMBEDDING: {:?}", e);
```
Comment (flvndvd) [inline core/src/providers/mistral.rs:1106]: ✂️

**Comment by spolu** [comment]
(Updated dev setup with a dev mistral api key)

**Comment by spolu** [comment]
(Also improved error handling as Mistral can return non JSON errors)

**Review by flvndvd — APPROVED** [review]

File: core/src/providers/mistral.rs [hunk 4]
```diff
@@ -959,6 +965,175 @@ impl LLM for MistralAILLM {
...
+                        "MistralAIError: failed parsing embeddings from Mistral AI err={} data={}",
+                        err,
+                        String::from_utf8_lossy(c),
+                    ))?,
+                };
+                unreachable!()
+            }
+        };
+
+        assert!(embeddings.data.len() > 0);
```
[the remark this answers is not in the document]
Comment (spolu) [reply core/src/providers/mistral.rs:1122]: This is cargo culted and I believe impossible in practice. We can refactor the 3 providers same as above. Leaning towards keeping it for consistency

**Pull request merged**